src/main/java/edu/umich/its/cpm/EmailFormatter.java

```diff
@@ -41,58 +58,189 @@ public EmailFormatter(String emailMgs, AttachmentHandler attachmentHandler) thro
+        String attachmentRemovedText = subStringBefore(rfcFormattedText, "Content-Transfer-Encoding: base64");
+        String[] msgBodyWithSomeExtra = attachmentRemovedText.split(NEW_LINE);
+        List<String> emailTextList = new LinkedList(Arrays.asList(msgBodyWithSomeExtra));
+        int size = emailTextList.size();
+        emailTextList.remove(size - 1); // remove the last element in the list
+        String lastItemInList = emailTextList.get(emailTextList.size() - 1) + "--"; // the replacing the new last element with new string
+        emailTextList.set(emailTextList.size() - 1, lastItemInList);
+        for (int i = 0; i < emailTextList.size(); i++) {
+            if (i == (emailTextList.size() - 2)) {
+                String lastLineInBody = emailTextList.get(i);
+                String appendToBodyText = NEW_LINE + NEW_LINE +
+                        "YOUR ATTACHMENTS ARE DROPPED DUE TO SIZE LIMIT";
+                emailTextList.set(i, lastLineInBody + appendToBodyText);
+            }
+            emailTextWithBodyNoAttachments.append(emailTextList.get(i));
+            emailTextWithBodyNoAttachments.append(NEW_LINE);
+        }
+
+
+        return emailTextWithBodyNoAttachments.toString();
     }
 
     /*
       Mbox Format standard is defined by RFC4155 document and extends RFC2822 Standard.
       Mbox format email messages file must start like "From doe@eg.edu Wed Aug 24 14:04:47 2016" and
       each message is separated by blank line
      */
-    public String mboxFormat() {
+    public List<Object> mboxFormat() {
         ArrayList<String> headers = mboxFormatHeaders();
-        return getMboxFormattedEmailText(headers);
+        List<Object> mboxFormatEmailTextPlusStatus = getMboxFormattedEmailText(headers);
+        String emailText = (String) mboxFormatEmailTextPlusStatus.get(0);
+        JSONObject status = (JSONObject) mboxFormatEmailTextPlusStatus.get(1);
+        if (emailText == null) {
+            status.put(Utils.JSON_ATTR_ITEM_STATUS, Utils.STATUS_ERROR);
+            status.remove(Utils.JSON_ATTR_MAIL_ATTACHMENTS);
+            mboxFormatEmailTextPlusStatus.set(1, status);
+            log.error("Mbox formatting is not successful");
+            return mboxFormatEmailTextPlusStatus;
+        }
+
+        JSONArray attachments = status.getJSONArray("attachments");
+        for (int i = 0; i < attachments.length(); i++) {
+            JSONObject attach = attachments.getJSONObject(i);
+            if(attach.get(Utils.MIGRATION_STATUS)!= Utils.STATUS_OK){
+                status.put(Utils.JSON_ATTR_ITEM_STATUS, Utils.STATUS_PARTIAL);
+                status.put(Utils.JSON_ATTR_MESSAGE,"Attachments URL from Ctools couldn't be resolved");
+                status.remove(Utils.JSON_ATTR_MAIL_ATTACHMENTS);
+                mboxFormatEmailTextPlusStatus.set(1, status);
+                return mboxFormatEmailTextPlusStatus;
+            }
+        }
+        // everything went fine at this point
```

Comment:
The email formatting went fine with no error (like attachment URL not resolved, we drop the attachment but don't drop the entire msg) or exception. IN short a pure  Success case!